Reorganize snippets into subobjects for each page

ethanbb opened the pull request:
As a first step to #513, this reorganizes `en.js` and the other language files somewhat so that each step is in its own nested object. The key matches the step title as specified by `VisitPage.js`, and this is used to pass only the snippets that are needed to each step (and likewise for HomePage and Footer). I also alphabetized the languages and made a couple aesthetic changes.

knod commented:
It would add consistency in general - each page being represented by a key. I'm not completely sure that `VisitPage` isn't significant, but it was kinda nice to have just one level of nesting. I know we haven't done anything with that yet, but it's often useful to prefer flat over deep as far as architecture goes.

Honestly, I'm torn on which way to jump. I'm leaning more towards consistency because I've seen inconsistency be a big barrier to bringing new folks on, but that may just be a story I've built in my head and I'm not sure the structure you have here lacks clarity. Maybe we can call out some of our members who are new to coding to pipe in on Slack about what they'd find least blocking.

knod commented:
It looks like a nested object makes sense to folks. Let's go with that for now.

ethanbb commented:
OK, I added `visitPage`. Also, separated the names of steps from their keys in the translation files in case we want to change the names in the future.  Finally I removed the quotes from the keys in the translation files for consistency, and since they should be valid identifier names anyway I would think.

knod commented on src/containers/VisitPage.js:
```diff
@@ -72,24 +72,29 @@ class VisitPage extends Component {
       {
-        title: 'Current Benefits',
-        form:  CurrentBenefitsStep,
+        displayName: 'Current Benefits',
```
Not blocking, but this may be better as `tabName` if we're going to change the prop name. Is this related to the snippets changes?

knod replied:
Forget this one, look at the next one instead. I just saw your note about what you were trying to do with this. I'm still not sure it's part of the snippet changes, but meh.

ethanbb replied:
We can keep it as `title` actually...I changed it to try to make it more clear that this was a display string as opposed to an internal identifier, but that's probably not necessary.

Although we could actually probably get rid of this field entirely, and instead add the title to the snippets for each step and use `snippets[ step.key ].title` for the tab text.

ethanbb replied:
Or `tabTitle` to keep it unambiguous

ethanbb replied:
Actually I guess it would be better to have all the tab names together in one section of the snippets

knod commented on src/containers/VisitPage.js:
```diff
@@ -71,7 +71,31 @@ class VisitPage extends Component {
     this.steps = [
-      { title: 'Current Benefits', form: CurrentBenefitsStep }, { title: 'Household', form: HouseholdStep }, { title: 'Income', form: CurrentIncomeStep }, { title: 'Expenses', form: CurrentExpensesStep }, { title: 'Predictions', form: PredictionsStep },//,
+      {
+        displayName: 'Current Benefits',
```
Not a blocking note, but this might be better called `tabName` or `tabTitle`.
